### PR TITLE
Secured Deep Learning in Remote Devices

### LINKED ISSUE #644: Secured Deep Learning in Remote Devices
### Brief Summary: 
Describe the what, why, and how of your content idea in 2-5 sentences.

In this article, we will understand how Secured deeplearning models can be deployed in Remote devices using Federated learning. We will also be going over a step by step guide for implementation.

### Key Takeaways: 
What are the 3-5 most important things the reader should understand or be able to do after reading this article?

- The reader will understand how deeplearning models are deployed in remote devices
- The reader will be able to code a simple deeplearning model, which can be deployed on remote devices
- The reader will understand the importance of Federated learning

### References: 
Please list links to any published content/research that you intend to use to support/guide this article. (If none, please indicate 'N/A'.)

- https://www.udacity.com/course/secure-and-private-ai--ud185
- https://ai.googleblog.com/2017/04/federated-learning-collaborative.html
- https://blogs.nvidia.com/blog/2019/10/13/what-is-federated-learning/

### PR DESCRIPTION
**Pull request by srishilesh**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      

#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

This fixes #644

### CODE REVIEW

**Comment by hectorkambow** [comment]
👋  Thank you for submitting this PR - we will review it ASAP - all PRs go into our queue will be looked over before being approved to be published.

**Comment by lalith1403** [comment]
Good morning @hectorkambow . The front matter is included. Just checking if you added it?

**Comment by hectorkambow** [comment]
> Good morning @hectorkambow . The front matter is included. Just checking if you added it?

No Sir - I did not. 👀

**Comment by hectorkambow** [comment]
plagiarism check done

File: articles/secured-deeplearning-in-remote-devices/index.md [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+  train_distributed_dataset = [] # Declare a new list
+  for batch_idx, (data,target) in enumerate(train_loader): # Load the data and target from the train dataset
+    data = data.send(compute_nodes[batch_idx % len(compute_nodes)]) # Separate the independent values from the train dataset
+    target = target.send(compute_nodes[batch_idx % len(compute_nodes)]) # Separate the target values from the train dataset
+    remote_dataset[batch_idx % len(compute_nodes)].append((data, target))
+```
+
+Here, `batch_idx % len(compute_nodes)` helps us index the `remote_dataset`. For our example, the index is `0` and `1`.
+
+#### Initializing Neural networks for each remote device
```
Comment (lalith1403) [inline articles/secured-deeplearning-in-remote-devices/index.md:172]: Title case.

File: articles/secured-deeplearning-in-remote-devices/index.md [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+Virtual workers are entities present on our local machine. They are used to model the behavior of actual workers. Then, we create 2 different workers for the demonstration.
+
+```python
+  hook = sy.TorchHook(torch) # Bind the tensor with local workers
+  end_device1 = sy.VirtualWorker(hook, id="device1") # 1st virtual entity
+  end_device2 = sy.VirtualWorker(hook, id="device2") # 2nd virtual entity
+  compute_nodes = [end_device1, end_device2] # List of workers
+```
+
+#### Distributing the Training Dataset to each Worker
```
Comment (lalith1403) [inline articles/secured-deeplearning-in-remote-devices/index.md:158]: title case

File: articles/secured-deeplearning-in-remote-devices/index.md [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+
+```bash
+Out[2]:
+Parameter containing:
+tensor([-0.0982], requires_grad=True)
+```
+
+We see that `device1` has a bias of `-0.0842`, and `device2` has a bias of `-0.0982`.
+
+#### Function for Model training
```
Comment (lalith1403) [inline articles/secured-deeplearning-in-remote-devices/index.md:218]: title case

**Review by lalith1403 — CHANGES_REQUESTED** [review]
@srishilesh An amazing article. I loved experimenting with the code. Some minor modifications and this should be good to go. Well done. 

The minor modifications need to be performed throughout the article. I haven't specified all the instances to keep the number of comments concise. Find all such occurrences and change it accordingly.

File: articles/secured-deeplearning-in-remote-devices/index.md [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+- [Installation](#installation)
+- [Implementation](#importing-libraries)
+- [Conclusion](#conclusion)
+- [Further Reading](#further-reading)
+
+### Introduction
+#### What is Federated Learning?
+In Deep Learning, the problem of privacy arises with the centralization of the data used in training and development. The nature of data is to remain private, accessible only to the end-users, not even the organization that is providing the service. But nowadays, we are unsure if our privacy is at stake.
+
+Generally, in any end-user device using Deep learning, the data is sent to the cloud, the predictions/classifications are made, and the results are returned to the end-users. Here, as end-users, we send our data to the cloud. There is no guarantee that our data is secure. That's when the idea of Federated learning (Distributed deep learning), came into the picture, to preserve privacy.
```
[the remark this answers is not in the document]
Comment (srishilesh) [reply articles/secured-deeplearning-in-remote-devices/index.md:34]: Sure. Thanks for the suggestion

**Comment by srishilesh** [comment]
Hi @lalith1403 . Thanks for the review. I have made the requested changes. Please let me know, if I have missed out any.

**Comment by hectorkambow** [comment]
@srishilesh 👋  Thank you for submitting your "Secured Deep Learning in Remote Devices" learn a bit about Deep learning in remote devices - please be sure to go over the changes I made to your article to be sure to implement any similar changes moving forward to other articles 🚀  
Fixed some sentences that were a bit difficult to understand (FYI) That may lead to a tiered payout
Thank you!

**Pull request merged**